uber/badge_funcs.py

```diff
@@ -98,6 +98,66 @@ def check_unassigned():
 # TODO: perhaps a check_leaderless() for checking for leaderless groups, since those don't get emails
 
 
+# run through all badges and check 2 things:
+# 1) there are no gaps in badge numbers
+# 2) all badge numbers are in the ranges set by c.BADGE_RANGES
+# note: does not do any duplicates checking, that's a different pre-existing check
+def badge_consistency_check(session):
+    errors = []
+
+    # check 1, see if anything is out of range, or has a duplicate badge number
+    badge_nums_seen = []
+
+    attendees = session.query(Attendee)\
+        .filter(Attendee.first_name != '')\
+        .filter(Attendee.badge_num != 0)\
+        .order_by('badge_num')\
+        .all()
+
+    for attendee in attendees:
+        out_of_range_error = check_range(attendee.badge_num, attendee.badge_type)
+        if out_of_range_error:
+            msg = '{a.full_name}: badge #{a.badge_num}: {err}'.format(a=attendee, err=out_of_range_error)
+            errors.append(msg)
+
+        if attendee.badge_num in badge_nums_seen:
+            msg = '{a.full_name}: badge #{a.badge_num}: Has been assigned the same badge number ' \
+                  'of another badge, which is not supposed to happen'.format(a=attendee)
+            errors.append(msg)
+
+        badge_nums_seen.append(attendee.badge_num)
+
+    # check 2: see if there are any gaps in each of the badge ranges
```

Comment:
It's totally fine to include this, but keep in mind that this will throw TONS of errors for attendee badges during/after fest because their numbers are pretty much assigned willy-nilly. Also, we don't have to police gaps the way we used to, because the new number auto-assign will 'fill in' gaps. (see: https://github.com/magfest/ubersystem/blob/master/uber/models.py#L582).